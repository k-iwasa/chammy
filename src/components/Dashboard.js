import React, { useState } from "react";
import { TextField } from "@material-ui/core";

export default function Dashboard() {
  const [name, setName] = useState("");

  const handleChange = (e) => {
    try {
      if (e.target.value === "") {
        throw new Error("湯婆婆エラー！！");
      }
      setName(e.target.value);
    } catch (error) {
      console.log("湯婆婆エラー出てるよ！！大変たいへん！困っちゃうよ！ぴえん");
    }
  };

  const newName = name.substr(Math.floor(Math.random() * name.length), 1);

  return (
    <>
      <div>契約書だよ。そこに名前を書きな</div>
      <TextField variant="outlined" onChange={handleChange} label="名前" />
      <div>フン。 {name}というのかい。贅沢な名だねぇ。</div>
      <div>
        今からお前の名は{newName}だ。いいかい、{newName}
        だよ。わかったら返事をするんだ、{newName}!!
      </div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
      <div>aa</div>
    </>
  );
}
