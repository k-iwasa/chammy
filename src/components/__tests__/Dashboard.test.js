import Dashboard from "../Dashboard";
import React from "react";
import userEvent from "@testing-library/user-event";
import { render } from "@testing-library/react";

describe("<Dashboard/>", () => {
  it("react testing librarya", () => {
    const { baseElement } = render(<Dashboard />);

    expect(baseElement).toMatchSnapshot();
  });

  it("react testing library clicka", () => {
    const { baseElement, getByText } = render(<Dashboard />);

    userEvent.click(getByText("clickme"));

    expect(baseElement).toMatchSnapshot();
  });
  it("react testing library 閉じる", () => {
    const { baseElement, getByText } = render(<Dashboard />);

    userEvent.click(getByText("clickme"));

    userEvent.click(getByText("閉じる"));

    expect(baseElement).toMatchSnapshot();
  });
  it("react testing library 閉じるほげ", () => {
    const wrapper = render(<Dashboard />);

    userEvent.click(wrapper.getByText("clickme"));

    userEvent.click(wrapper.getByText("閉じる"));

    expect(wrapper.baseElement).toMatchSnapshot();
  });
});
