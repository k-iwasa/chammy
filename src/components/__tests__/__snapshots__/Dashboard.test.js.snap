// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard/> react testing library clicka 1`] = `
<body
  style="padding-right: 0px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        clickme
      </span>
      <span
        class="MuiTouchRipple-root"
      >
        <span
          class="MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
          style="width: 2.8284271247461903px; height: 2.8284271247461903px; top: -1.4142135623730951px; left: -1.4142135623730951px;"
        >
          <span
            class="MuiTouchRipple-child MuiTouchRipple-childLeaving"
          />
        </span>
      </span>
    </button>
    <div>
      2
    </div>
  </div>
  <div
    class="MuiDialog-root"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper"
      role="none presentation"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        class="MuiPaper-root MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiPaper-elevation24 MuiPaper-rounded"
        role="dialog"
      >
        <div>
          hoge
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            閉じる
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`<Dashboard/> react testing library 閉じる 1`] = `
<body
  style="padding-right: 0px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        clickme
      </span>
      <span
        class="MuiTouchRipple-root"
      >
        <span
          class="MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
          style="width: 2.8284271247461903px; height: 2.8284271247461903px; top: -1.4142135623730951px; left: -1.4142135623730951px;"
        >
          <span
            class="MuiTouchRipple-child MuiTouchRipple-childLeaving"
          />
        </span>
      </span>
    </button>
    <div>
      2
    </div>
  </div>
  <div
    class="MuiDialog-root"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root"
      style="opacity: 0; webkit-transition: opacity 195ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 195ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper"
      role="none presentation"
      style="opacity: 0; webkit-transition: opacity 195ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 195ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        class="MuiPaper-root MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiPaper-elevation24 MuiPaper-rounded"
        role="dialog"
      >
        <div>
          hoge
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            閉じる
          </span>
          <span
            class="MuiTouchRipple-root"
          >
            <span
              class="MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
              style="width: 2.8284271247461903px; height: 2.8284271247461903px; top: -1.4142135623730951px; left: -1.4142135623730951px;"
            >
              <span
                class="MuiTouchRipple-child MuiTouchRipple-childLeaving"
              />
            </span>
          </span>
        </button>
      </div>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`<Dashboard/> react testing library 閉じるほげ 1`] = `
<body
  style="padding-right: 0px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        clickme
      </span>
      <span
        class="MuiTouchRipple-root"
      >
        <span
          class="MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
          style="width: 2.8284271247461903px; height: 2.8284271247461903px; top: -1.4142135623730951px; left: -1.4142135623730951px;"
        >
          <span
            class="MuiTouchRipple-child MuiTouchRipple-childLeaving"
          />
        </span>
      </span>
    </button>
    <div>
      2
    </div>
  </div>
  <div
    class="MuiDialog-root"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root"
      style="opacity: 0; webkit-transition: opacity 195ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 195ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper"
      role="none presentation"
      style="opacity: 0; webkit-transition: opacity 195ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 195ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        class="MuiPaper-root MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiPaper-elevation24 MuiPaper-rounded"
        role="dialog"
      >
        <div>
          hoge
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            閉じる
          </span>
          <span
            class="MuiTouchRipple-root"
          >
            <span
              class="MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
              style="width: 2.8284271247461903px; height: 2.8284271247461903px; top: -1.4142135623730951px; left: -1.4142135623730951px;"
            >
              <span
                class="MuiTouchRipple-child MuiTouchRipple-childLeaving"
              />
            </span>
          </span>
        </button>
      </div>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`<Dashboard/> react testing librarya 1`] = `
<body>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        clickme
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <div>
      1
    </div>
  </div>
</body>
`;
